### ISSUE TITLE
no-unused-prop-types false positive with createPortal

### ISSUE THREAD

**Issue by reintroducing** [jsx-eslint/eslint-plugin-react#3475]
I've got a component where the containing div is wrapped with `createPortal`.  If I remove the createPortal, all of my propTypes are correctly used, but if I re-add the createPortal to surround the div, all of my propTypes get marked as no-unused-prop-types. Not sure whats going on here but figured I'd point it out in case nobody has ever reported this (which doesn't seem anyone has, I searched both open and closed issues with no success around this).

**Comment by himanshu007-creator** [comment]
Hey @reintroducing , can you share a codespace sample? like a stackblitz link?
Reference: https://stackblitz.com/

**Comment by reintroducing** [comment]
@himanshu007-creator Is there a pre-existing template set up already with the rules in place for this repo? Or do I have to create a new one to demonstrate this issue?

**Comment by himanshu007-creator** [comment]
You can create a blank project to demonstrate the code that you are referring to.

**Comment by reintroducing** [comment]
@himanshu007-creator right, i guess what i meant is if there is a pre-existing template that takes care of the setup of defining eslint, adding this plugin, and being ready to show any subsequent issues that may be present, but i'm assuming there isn't, so i'll create one from scratch.

**Comment by himanshu007-creator** [comment]
Yeah you would have to create a fresh one. Sorry for my misinterpretation 🙂

**Comment by reintroducing** [comment]
Here you go: https://stackblitz.com/edit/vitejs-vite-xkspmk

I updated the title because its actually when createPortal is used within a forwarded ref. `FooNoPortal.jsx` and `FooPortal.jsx` are identical except that the NoPortal doesn't create a portal and the other does. The one with the forwarded ref and the portal produces the error in the terminal:

`src/FooPortal.jsx(10,3): 'name' PropType is defined but prop is never used react/no-unused-prop-types`

**Comment by reintroducing** [comment]
@himanshu007-creator just following up to see if the SB link above worked for you and you are able to verify this is a real issue?

**Comment by giovannipds** [comment]
We're facing false positives in our project too

**Comment by afzaalahmad** [comment]
I am facing the same issue with forwardRef in a higher order component that I've written.

My hoc is like this:
```
import type React from 'react';
import type { ObjectSchema } from 'yup';
import type { UseFormProps, UseFormReturn, FieldValues, DefaultValues } from 'react-hook-form';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { forwardRef, memo, useImperativeHandle } from 'react';
import { useTranslation, UseTranslationResponse } from 'react-i18next';

export type Context = 'CREATE' | 'UPDATE';

export interface Props<T extends FieldValues> {
  // it is complaining on all of these fields
  context?: Context;
  readonly?: boolean;
  initialValues?: DefaultValues<T>;
}

export type FormRefAttributes<T extends FieldValues, C = unknown> = Pick<
  UseFormReturn<T, C>,
  'handleSubmit' | 'watch' | 'setError' | 'formState'
>;

export const withFormValidation = <T extends FieldValues, P = unknown, C = unknown>(
  Form: React.FC<P & Props<T> & UseFormReturn<T, C>>,
  getFormSchema: (context: Context, t: UseTranslationResponse<any>) => ObjectSchema<T>,
  useFormProps?: Omit<UseFormProps<T, C>, 'resolver'>,
): React.MemoExoticComponent<
  React.ForwardRefExoticComponent<React.PropsWithoutRef<P & Props<T>> & React.RefAttributes<FormRefAttributes<T, C>>>
> =>
  memo(
    forwardRef<FormRefAttributes<T, C>, P & Props<T>>((props, ref) => {
      const { context = 'CREATE', initialValues } = props;
      const translation = useTranslation();
      const form = useForm<T, C>({
        mode: 'onSubmit',
        reValidateMode: 'onChange',
        defaultValues: initialValues,
        resolver: yupResolver<ObjectSchema<T | any>>(getFormSchema(context, translation)),
        ...useFormProps,
      });

      useImperativeHandle(ref, () => form, [form]);
      return <Form context={'CREATE'} {...props} {...form} />;
    }),
  );
```
My usage of hoc is
```
export const ContactDetailsForm = withFormValidation<ContactFields>((props) => <Form {...props} />)
```

and then I am using this form as 
```
<ContactDetailsForm initialValues={initialData} readonly={true} />
```

Now I am using these props, I just joined them using & with generics of hoc.

Let me know if you have any questions.
